src/pages/membership/components/youthProfileForm/YouthProfileForm.tsx

```diff
@@ -45,30 +41,34 @@ const schema = Yup.object().shape({
   approverFirstName: Yup.string()
     .min(2, 'validation.tooShort')
     .max(255, 'validation.tooLong')
-    .when([], (schema: Yup.StringSchema) => {
-      return AGE < ageConstants.ADULT
+    .when(['birthDate'], (value: string, schema: Yup.StringSchema) => {
```

Comment:
You have the same check (if I'm reading correctly) copied to three places. Would it make sense to split that out to a separate boolean, so we could have something like `when(isConsentRequired)` ?

Reply:
Because schema is outside of the component, saving value to boolean would require adding it to form values, which on it's own way doesn't sound like a good idea. 

I could change it to something like this:
```suggestion
    .when(['birthDate'], (value: string, schema: Yup.StringSchema) => return isConsentRequired(birthDate)
    
    const isConsentRequired = (birthDate:string) => {
      const userAge = differenceInYears(new Date(), new Date(value));
      return userAge < ageConstants.ADULT
        ? schema.required('validation.required')
        : schema;
    }
```
Creating a custom validation method would be other option, but doing that seems like a hassle with TypeScript i'm not sure if it's possible to access other fields value ([example](https://github.com/jquense/yup/issues/312#issuecomment-442854307))

Reply:
This was exactly what I meant! 

In case the rules for consent changes we only need to update the logic in one single place.